src/Commands/Extension/AzCommand.cs

```diff
@@ -105,6 +106,54 @@ protected override AzArguments BindArguments(ParseResult parseResult)
         return null;
     }
 
+    private static async Task<bool> AuthenticateWithAzureCredentialsAsync(IExternalProcessService processService, ILogger logger)
```

Comment:
@christothes - Does Azure Identity support AZURE_CREDENTIALS?  I know we have moved away from this, but in some cases this is the only option for devs - container dev.    Or should we switch to asking them to use AZURE_CLIENT_ID, TENANT_ID, CLIENT_SECRET instead of AZURE_CREDENTIALS?

Reply:
I'm not aware of any support we have for AZURE_CREDENTIALS

Reply:
If we do decided to use AZURE_CLIENT_ID, TENANT_ID, CLIENT_SECRET, we could add EnvironmentCredential to the chain and it would get picked up automatically.

Reply:
It looks like AZURE_CREDENTIALS is something started in GitHub Actions. Is there a reason we want to support that pattern here?

Reply:
The goal is to have people authenticate to Azure CLI using it, like in the Azure CLI MCP documentation at https://github.com/jdubois/azure-cli-mcp?tab=readme-ov-file#install-and-configure-the-server-with-docker (on which this was copied)

Reply:
@jdubois - Can you create an AzureCredentialsEnvironmentCredential that derives from TokenCredential in Azure.Idenitty?

This should be a good starting point: https://github.com/Azure/azure-sdk-for-net/blob/8e101ec21e839ddb846b95500c214013842396dd/sdk/identity/Azure.Identity/src/Credentials/EnvironmentCredential.cs#L51

Reply:
@christothes 
Looking at impl in PR.
Details from AZURE_CREDENTIALS are used to auth as service principal via az cli.

Can we add this support to az cli credential to consume AZURE_CREDENTIALS file and auth as service principal ?

Reply:
Is there a file path or is the content of the json set as the value of the AZURE_CREDENTIALS env var? I think for now we should wrap the below impl in a local instance of TokenCredential. We should consider if this should be added to the existing AzureCliCredential separately.

Reply:
I agree, let's get it going in the MCP repo as a custom token credential and then see what the identity team thinks.

Reply:
Okay, I will open a PR to add this as a custom token credential to the MCP repo.

Reply:
synced with @christothes, he will open a PR to add this in MCP repo.

Reply:
@jongio I started to look at this more closely and I'm not sure we need an implementation of `TokenCredential`. The reason is that it seems that we just want to ensure that `az` is logged in with the credentials in AZURE_CREDENTIALS but we don't need the actual token. If that is true, what we have here seems appropriate.

Reply:
Once we introduce the abstraction of Cached Token Credential to optimize and provide in memory caching. 
This layer won't be able to inherit it. 
Wrapping it in Token Credential also allows to manage auth centrally in MCP services.

Reply:
synced with @christothes 
There is no need for TokenCredential abstraction here. 
We just need to abstract out the impl into AzureCLILoginProvider under auth package.

Reply:
Sounds good.  I'll let @g2vinay and @christothes work out a good plan